src/sentry/workflow_engine/types.py

```diff
@@ -32,17 +33,19 @@ class DetectorPriorityLevel(IntEnum):
 ProcessedDataConditionResult = tuple[bool, list[DataConditionResult]]
 
 
-class EventJob(TypedDict):
+@dataclass(frozen=True)
+class EventJob:
     event: GroupEvent
 
 
-class WorkflowJob(EventJob, total=False):
-    group_state: GroupState
-    is_reprocessed: bool
-    has_reappeared: bool
-    has_alert: bool
-    has_escalated: bool
-    workflow: Workflow
+@dataclass(frozen=True)
+class WorkflowJob(EventJob):
+    group_state: GroupState | None = None
+    is_reprocessed: bool | None = None
+    has_reappeared: bool | None = None
+    has_alert: bool | None = None
+    has_escalated: bool | None = None
+    workflow: Workflow | None = None
```

Comment:
I guess it's not possible to create this with all these attributes set when we first create it? I think that ideally we'd make as many of these required as possible

Reply:
the `event` is always required, for example if we call `process_workflows` for metric detectors. but that flow doesn't go through post process so it won't have these extra fields that i had to mark as optional

Reply:
I'm wondering if maybe we should have different datastructures for these different places, maybe something to keep in mind as you work on things